apps/src/studio/Item.js

```diff
@@ -358,3 +444,37 @@ Item.prototype.moveToNextPosition = function () {
   this.x = next.x;
   this.y = next.y;
 };
+
+/**
+ * Mark that we're colliding with object represented by key.
+ * Here, override base implemention to special on-collision logic for certain
+ * item classes.
+ * @param key A unique key representing the object we're colliding with
+ * @returns {boolean} True if collision is started, false if we're already colliding
+ * @override
+ */
+Item.prototype.startCollision = function (key) {
+  var newCollisionStarted = Item.superPrototype.startCollision.call(this, key);
+  if (newCollisionStarted) {
+    if (this.className === 'hazard') {
```

Comment:
I like that, but it doesn't give level builders quite as much flexibility. But seems like an improvement for the time being. And it makes Item.js unaware of hard-coded class names.